src/renderers/dom/shared/devtools/ReactDOMDuplicateSelectValuesDevtool.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactDOMDuplicateSelectValuesDevtool
+ */
+
+'use strict';
+
+var ReactComponentTreeDevtool = require('ReactComponentTreeDevtool');
+var warning = require('warning');
+var areEqual = require('areEqual');
+
+var didWarnDupeSelectValues = false;
+
+function handleElement(debugID, element) {
+  
+  if (didWarnDupeSelectValues) {
+    return;
+  }
+  
+  if (element == null) {
+    return;
+  }
+
+  if (element.type !== 'select') {
+    return;
+  }
+
+  if ((element.props == null) || (element.props.children == null)) {
+    return;
+  }
+
+  // Uncontrolled select elements can have duplicate values.
+  if (element.props.hasOwnProperty('defaultValue')) {
+    return;
+  }
+
+  let values = [];
+  let options = element.props.children;
+
+  //  If options is not iterable make it an array.
+  if (typeof options[Symbol.iterator] !== 'function') {
+    options = [options];
+  }
+
+  // Combine the values from all options into a single array.
+  for (const option of options) {
+    if (option.type === 'optGroup') {
+      if ((option.props != null) && (option.props.children != null)) {
+        let groupOptions = option.props.children;
+        //  If groupOptions is not iterable make it an array.
+        if (typeof groupOptions[Symbol.iterator] !== 'function') {
+          groupOptions = [groupOptions];
+        }
+        for (const groupOption of groupOptions) {
+          if ((groupOption.props != null) && (groupOption.props.value != null)) {
+            values.push(groupOption.props.value);
+          }
+        }
+      }
+    }
+
+    if (option.type === 'option') {
+      if ((option.props != null) && (option.props.value)) {
+        values.push(option.props.value);
+      }
+    }
+  }
+
+  if (values.length <= 1) {
+    return;
+  }
+
+  // Check the array for duplicate values.
+  for (var i = 0; i < values.length-1; i++) {
```

Comment:
Thanks for the feedback. I agree that this is not the fastest way to compare arrays. The `value` dictionary wouldn't work because `value` can be an object, array, etc. I thought of using a dictionary of Adler32 checksums but that seemed like overkill for this.